R/retrieveData.R

```diff
@@ -26,166 +27,230 @@
-    inargs[toadd] <- defargs[toadd]
-  }
-  if (length(inargs) > 0) {
-    hashs <- digest(inargs, algo = getConfig("hash"))
-    if (uselabels) {
-      hashs <- toolCodeLabels(hashs)
+    if (bundle) {
+      matchingBundles <- .match(getConfig("bundlefolder"), "bdl", cfg$bundleName)
+
+      if (length(matchingBundles) == 1) {
+        vcat(-2, " - data will be created from existing bundle (", matchingBundles, ").", fill = 300)
+        do.call(bundleCompile, c(list(bundle = file.path(getConfig("bundlefolder"), matchingBundles)),
+                                 cfg$input[cfg$bundleArguments]))
+        return()
+      }
     }
-    argsHash <- paste0(hashs, "_")
-  } else {
-    argsHash <- NULL
-  }
 
-  regionmapping <- getConfig("regionmapping")
-  if (!file.exists(regionmapping)) {
-    regionmapping <- toolGetMapping(regionmapping, type = "regional", returnPathOnly = TRUE)
   }
-  regionscode <- regionscode(regionmapping, label = uselabels)
 
-  rev <- numeric_version(rev)
+  # data not yet ready and has to be prepared first
+  sourcefolder <- paste0(getConfig("outputfolder"), "/", cfg$collectionName)
 
-  collectionname <- paste0(
-    "rev", rev, dev, "_", regionscode, "_", argsHash, tolower(model),
-    ifelse(getConfig("debug") == TRUE, "_debug", "")
-  )
-
-  matchingCacheFiles <- dir(path = getConfig("outputfolder"), pattern = ".*\\.tgz")
-  matchingCacheFiles <- matchingCacheFiles[startsWith(matchingCacheFiles, collectionname)]
-  if (!isTRUE(getConfig("debug"))) { # do not use debug files if not in debug mode
-    matchingCacheFiles <- matchingCacheFiles[!startsWith(matchingCacheFiles, paste0(collectionname, "_debug"))]
+  # create folder if required
+  if (!file.exists(sourcefolder)) {
+    dir.create(sourcefolder, recursive = TRUE)
   }
-  if (length(matchingCacheFiles) == 0 || getConfig("debug") == TRUE) {
-    # data not yet ready and has to be prepared first
-    sourcefolder <- paste0(getConfig("outputfolder"), "/", collectionname)
 
-    # create folder if required
-    if (!file.exists(sourcefolder)) {
-      dir.create(sourcefolder, recursive = TRUE)
+  # copy mapping to mapping folder and set config accordingly
+  regionmapping <- toolGetMapping(getConfig("regionmapping"), type = "regional", returnPathOnly = TRUE)
+  mappath <- toolGetMapping(paste0(cfg$regionscode, ".csv"), "regional", error.missing = FALSE, returnPathOnly = TRUE)
+  if (!file.exists(mappath)) {
+    if (!dir.exists(dirname(mappath))) {
+      dir.create(dirname(mappath), recursive = TRUE)
     }
+    file.copy(regionmapping, mappath)
+  }
+  # copy mapping to output folder
+  try(file.copy(regionmapping, sourcefolder, overwrite = TRUE))
+  try(saveRDS(list(package = attr(cfg$functionName, "package"),
+                   args = argumentValues[!(names(argumentValues) %in% names(cfg$setConfig))],
+                   bundleArguments = cfg$bundleArguments, sessionInfo = sessionInfo()),
+              file.path(sourcefolder, "config.rds"), version = 2))
```

Comment:
Maybe a warning in case of a failure in these try statements would be a good idea here?

Reply:
I realized try will still print the error to stdout/via `cat`, but I don't think that's good enough, because these messages are so easy to miss in a long log. Also, it's confusing to have a completely normal message starting with "Error: ...". If instead the error was converted to a warning both of these problems are solved, because by default warnings are printed at the end of the log (easy to find) and they are printed as "Warning: ..." instead of starting with "Error: ...". Furthermore our "warnings are not okay" policy should apply here which is not entirely clear if we would have a normal cat message.
tldr: Can we please use errorToWarning instead of try here? (And in all similar situations in this PR and in the future?)
```r
errorToWarning <- function(expr) {
  maybeError <- try(expr, silent = TRUE)
  if (inherits(maybeError, "try-error")) {
    warning(maybeError)
  }
}
```